org.abapgit.adt.ui/src/org/abapgit/adt/ui/internal/repositories/AbapGitView.java

```diff
@@ -167,7 +185,32 @@ public Image getImage(Object obj) {
 	 */
 	@Override
 	public void createPartControl(Composite parent) {
-		setupViewer(parent);
+		final Composite composite = new Composite(parent, SWT.NONE);
+		GridData gd = new GridData(SWT.FILL, SWT.FILL, true, true);
+		composite.setLayoutData(gd);
+		GridLayout layout = new GridLayout(1, false);
+		composite.setLayout(layout);
+
+		//Create Filter Text Box
+		this.searchText = new Text(composite, SWT.SEARCH | SWT.CANCEL);
+		this.searchText.setMessage(Messages.AbapGitView_Type_Filter_Text);
+		this.searchText.setLayoutData(new GridData(GridData.GRAB_HORIZONTAL | GridData.HORIZONTAL_ALIGN_FILL));
+
+		setupViewer(composite);
+
+		//Add Listener to the Filter Text Box
+		this.searchText.addModifyListener(new ModifyListener() {
+
+			@Override
+			public void modifyText(ModifyEvent e) {
+				AbapGitView.this.tableFilter.setSearchText(AbapGitView.this.searchText.getText());
+				AbapGitView.this.viewer.refresh();
```

Comment:
If the searchText.getText() = "" // empty string, 
are we setting the text in the filter box back to Messages.AbapGitView_Type_Filter_Text ?